src/download-artifact.ts

```diff
@@ -14,12 +18,43 @@ export const chunk = <T>(arr: T[], n: number): T[][] =>
     return acc
   }, [] as T[][])
 
+function checkIfNotFoundOption(if_not_found: string, message: string) {
+  switch (if_not_found) {
+    case NotFoundOptions.warn: {
+      core.warning(message)
+      break
+    }
+    case NotFoundOptions.error: {
+      core.setFailed(message)
+      break
+    }
+    case NotFoundOptions.ignore: {
+      core.info(message)
+      break
+    }
```

Comment:
This is over-engineered and yet not remotely helpful.

It means that anything that wants this information would have to have `actions:read` so that it could ask for the current logs and retrieve the information provided.

That's really roundabout.

I'd much sooner use `gh run download` than deal with such an awkward design.